services/jenkins/jenkins-plugin-installs.service.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+
+const Joi = require('joi')
+
+const BaseJsonService = require('../base-json')
+const { NotFound, InvalidParameter } = require('../errors')
+const {
+  downloadCount: downloadCountColor,
+} = require('../../lib/color-formatters')
+const { metric } = require('../../lib/text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schemaInstallations = Joi.object()
+  .keys({
+    installations: Joi.object()
+      .required()
+      .pattern(nonNegativeInteger, nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+const schemaInstallationsPerVersion = Joi.object()
+  .keys({
+    installationsPerVersion: Joi.object()
+      .required()
+      .pattern(Joi.string(), nonNegativeInteger)
+      .min(1),
+  })
+  .required()
+
+class JenkinsPluginInstalls extends BaseJsonService {
+  async fetch({ plugin, version }) {
+    const url = `https://stats.jenkins.io/plugin-installation-trend/${plugin}.stats.json`
+    const schema = this.constructor._getSchema(version)
+    return this._requestJson({ url, schema })
+  }
+
+  static render({ label, installs }) {
+    return {
+      label: label,
+      message: metric(installs),
+      color: downloadCountColor(installs),
+    }
+  }
+
+  static _getSchema(version) {
+    if (version) {
+      return schemaInstallationsPerVersion
+    } else {
+      return schemaInstallations
+    }
+  }
+
+  static _getLabel(version) {
+    if (version) {
+      return 'installs@' + version
+    } else {
+      return 'installs'
+    }
+  }
+
+  async handle({ info, plugin, version }) {
+    const label = this.constructor._getLabel(version)
+    const json = await this.fetch({ plugin, version })
+
+    let installs
+    if (info === 'iv') {
+      installs = json.installationsPerVersion[version]
+      if (!installs) {
+        throw new NotFound({
+          underlyingError: new Error('non-existent version'),
+        })
+      }
+    } else if (info === 'i') {
+      const latestDate = Object.keys(json.installations)
+        .sort()
+        .slice(-1)[0]
+      installs = json.installations[latestDate]
+    } else {
+      throw new InvalidParameter({
+        underlyingError: new Error('invalid info'),
+      })
+    }
+
+    return this.constructor.render({ label, installs })
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'installs' }
+  }
+
+  static get category() {
+    return 'downloads'
+  }
+
+  static get url() {
+    return {
+      base: 'jenkins/plugin',
+      format: '(i|iv)/([^/]+)/?([^/]+)?',
+      capture: ['info', 'plugin', 'version'],
+    }
+  }
+
+  static get examples() {
```

Comment:
I'm wondering about the URLs here. Wouldn't simply having `'i/view-job-filters` and `i/view-job-filters/1.26` be enough to indicate whether we're targeting the total or a specific version? In other words, do we really need to make the distinction with two path variants, `iv` and `i`?

Similarly, for a lot of build badges, we optionally support branches, and branches are simply indicated by appending something like `/master` to the standard URL.

In addition to being slightly more consistent with those badges, this will simplify the URL regex